resources/qml/Preferences/Materials/MaterialsSyncDialog.qml

```diff
@@ -429,10 +429,61 @@ UM.Window
                                 Cura.SecondaryButton
                                 {
                                     id: refreshListButton
+                                    property int accountState: Cura.API.account.syncState
+                                    readonly property int _AccountSyncState_SYNCING: 0
+                                    readonly property int _AccountSyncState_SUCCESS: 1
                                     Layout.alignment: Qt.AlignVCenter
                                     text: catalog.i18nc("@button", "Refresh List")
                                     iconSource: UM.Theme.getIcon("ArrowDoubleCircleRight")
                                     onClicked: Cura.API.account.sync(true)
+
+                                    onAccountStateChanged: {
+                                        if (Cura.API.account.syncState == _AccountSyncState_SYNCING)
+                                        {
+                                            refreshListButton.visible = false;
+                                        }
+                                        else if (Cura.API.account.syncState == _AccountSyncState_SUCCESS)
+                                        {
+                                            refreshListButton.visible = true;
+                                        }
```

Comment:
```suggestion
                                        refreshListButton.visible = accountState != _AccountSyncState_SYNCING   
```

The button just needs to be hidden if the syncing is happening, right? There are a few other states (like error), where the button should become visible again. 

That being said. I think you can even just use

`visible: Cura.API.account.syncState != _AccountSyncState_SYNCING    `

and get rid of the accountState property.

I'm not a 100% sure, but if i remember correctly, invisible buttons can still be clicked. If that's the case, you should also set the enabled property (enabled: visible)